storage/shard.go

```diff
@@ -199,6 +213,24 @@ func newDatabaseShard(
 	return d
 }
 
+func (s *dbShard) initRuntimeOptions(mgr RuntimeOptionsManager) {
```

Comment:
I wonder if it's better to make this more generic. The runtime option setup is generic enough so it can be used on other types as well, so perhaps we should have a `RuntimeUpdater` interface:
```
type RuntimeUpdater interface{
  RuntimeOptions() RuntimeOptions
  SetRuntimeOptions()
}
```
and have `dbShard` implement the interface